File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/ButtonColorShape.java
package coolpharaoh.tee.speicher.tea.timer.views.new_tea;

import android.app.Application;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;

import androidx.core.content.ContextCompat;

import static coolpharaoh.tee.speicher.tea.timer.core.tea.Variety.BLACK_TEA;

class ButtonColorShape {

    private final Application application;
    private final GradientDrawable gradientDrawable;
    private int color;

    ButtonColorShape(Drawable drawable, Application application) {
        gradientDrawable = (GradientDrawable) drawable;
        this.application = application;

        setDefaultColor();
    }

    private void setDefaultColor() {
        int varietyColor = ContextCompat.getColor(application, BLACK_TEA.getColor());
        setColor(varietyColor);
    }

    void setColor(int color) {
        this.color = color;
        gradientDrawable.setColor(color);
    }

    int getColor() {
        return color;
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/ButtonColorShape.kt
package coolpharaoh.tee.speicher.tea.timer.views.new_tea

import android.app.Application
import android.graphics.drawable.Drawable
import android.graphics.drawable.GradientDrawable
import androidx.core.content.ContextCompat
import coolpharaoh.tee.speicher.tea.timer.core.tea.Variety

internal class ButtonColorShape(drawable: Drawable, application: Application) {

    private val application: Application
    private val gradientDrawable: GradientDrawable

    init {
        gradientDrawable = drawable as GradientDrawable
        this.application = application

        setDefaultColor()
    }

    private fun setDefaultColor() {
        val varietyColor = ContextCompat.getColor(application, Variety.BLACK_TEA.color)
        color = varietyColor
    }

    var color = 0
        set(color) {
            field = color
            gradientDrawable.setColor(color)
        }
}